src/main/java/com/faforever/client/map/MapServiceImpl.java

```diff
@@ -92,14 +98,16 @@
   public MapServiceImpl(PreferencesService preferencesService, TaskService taskService,
                         ApplicationContext applicationContext,
                         FafService fafService, AssetService assetService,
-                        I18n i18n, UiService uiService, ClientProperties clientProperties) {
+                        I18n i18n, UiService uiService, ClientProperties clientProperties, MapGeneratorService mapGeneratorService, EventBus eventBus) {
     this.preferencesService = preferencesService;
     this.taskService = taskService;
     this.applicationContext = applicationContext;
     this.fafService = fafService;
     this.assetService = assetService;
     this.i18n = i18n;
     this.uiService = uiService;
+    this.mapGeneratorService = mapGeneratorService;
+    this.eventBus = eventBus;
```

Comment:
Try using spring eventbus

Reply:
Of course, should've commented that, was too lazy the first time, gonna change that soon ;).

Reply:
There is no Spring Eventbus. What do you mean?

Reply:
Need to look up bean name